src/preferences/general.cpp

```diff
@@ -541,7 +541,7 @@ std::string language()
 
 void set_language(const std::string& s)
 {
-	preferences::set("locale", s);
+	prefs["locale"] = s;
```

Comment:
Is `set` itself getting removed?

Reply:
That was something I was aiming for originally actually - get rid of the generic get()/set() methods and thereby force everything to create explicit getters/setters for the preference they want so that you don't need to go searching around the codebase to get a full list of preferences that exist. Unfortunately:
* The migrator needs the ability to write out arbitrary preference values (might be able to work around it by just having it directly write the re-migrated `preferences` file back out).
* The lua preferences API needs the ability to write out arbitrary preferences (no idea how to deal with that).
* The preferences source files apart from `general.[ch]pp` don't actually have direct access to the `prefs` config (thinking of a larger preferences handling refactor/consolidation to deal with that).

Reply:
Maybe you can keep the generic get/set but make them private? The downside of this would be that the preferences namespace would need to be turned into a class, which probably means merging all the preferences header files together (or maybe doing something weird like including files in the middle of a class), but it _would_ enable you to ensure that _only_ preferences internals, the migrator, and the Lua preferences API have direct access to those functions.

Reply:
Would declaring them as friend classes work here? I've never needed to use that feature before, but it sounds like it'd work here.

Reply:
I was thinking friend functions rather than classes, but yeah, something like that. I don't know much about the migrator, but in the Lua case it would probably be declaring two functions as friends. Maybe it would be an entire class for the migrator.

Reply:
Well, either way I'll take a closer look at what to do in a couple days after I merge this.